.github/workflows/ci.yml

```diff
@@ -0,0 +1,93 @@
+name: Check
+
+on:
+  push:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  check_duplicate_runs:
```

Comment:
There is an annoying bug of Github Actions - when someone creates a new branch and pushes some commits here, and then creates a pull request, there are 2 copies of workflow running - one for a push event and another for created PR.
This step prevents the second copy of workflow to be run - if tests are already passed for push event then they will be not run twice while PR is created. Also they will not be run if source code hasn't been changed.

Reply:
This is a neat idea to avoid duplicate work. I don't think we need it anymore though, given we have this:

```yaml
on:
   push:
   pull_request:
     branches:
       - master
```

that is, we'd only build if something is a PR or it's a push to master, we'd never trigger two runs for a PR branch.

Reply:
Well, if you don't use internal PRs from feature branch to a master, then this step is not needed.